src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -46,7 +47,8 @@ public class HtmlHelper : IHtmlHelper, IViewContextAware
             IModelMetadataProvider metadataProvider,
             IViewBufferScope bufferScope,
             HtmlEncoder htmlEncoder,
-            UrlEncoder urlEncoder)
+            UrlEncoder urlEncoder,
+            DiagnosticSource diagnosticSource)
```

Comment:
FWIW in other libs, we're going the clean route and changing the existing constructor since this it's a V2 release. Keep it clean since it's a good time to do so, IMO.